### ISSUE TITLE
Haxelib added twice

### ISSUE THREAD

**Issue by fzzr-** [TiVo/intellij-haxe#101]
I have a library (in my haxelib-repo) named "Signal". In the "Project Structure" I added from "Global Libraries" my library. And this library mansioned in `build.hxml`:

````hxml
-cp src
-cp tests
-main Main
-lib signal
-dce full
--each
-x out/tests/main
````

After reopen/reindex this project i get two similar dependencies, where one is auto-generated.
Need to test existing dependencies before creating new ones. And additionally try find it in "Global Libraries" or project's "Libraries".

![screen shot 2015-01-11 at 17 54 43](https://cloud.githubusercontent.com/assets/888526/5695560/0c2ba3d0-99bb-11e4-9189-8ccede4dc750.png)

**Comment by EBatTiVo** [comment]
There is logic to detect this already in place.  Obviously, it isn't working.   :/

Is the path that "Signal 1,0,0" is installed to the same as that found and placed in the auto-generated path?  (e.g. Is it installed to "/Users/ak/Developer/Haxe/libs/signal/1,0,0"?)  Or is it installed in two places on your machine -- your user space and a global installation?

**Comment by fzzr-** [comment]
Yes, "Signal 1,0,0" installed only to "/Users/ak/Developer/Haxe/libs/signal/1,0,0".
Path of Haxelib-repo is /Users/ak/Developer/Haxe/libs.

**Comment by as3boyan** [comment]
Ok, second copy - that comes from my original implementation, first "Signal 1.0" maybe added somewhere(by TiVo) not sure, need to test

**Comment by fzzr-** [comment]
Stop! This is misunderstanding. "Signal 1.0" I created early into the Global Libraries and added it to dependencies to module. For plugin need checking already existing dependencies before creating/adding new.

**Comment by as3boyan** [comment]
Oh, I see, we do not check global libraries currently, I haven't found a way to do this yet. Good idea.

Ok, then I read title I thought that plugin adds it twice.

**Comment by fzzr-** [comment]
One place, one path, but two names:
1. "Signal 1.0" - it's i handy added;
2. name=path - generated by plugin.
Both have a path =  "/Users/ak/Developer/Haxe/libs/signal/1,0,0".

**Comment by as3boyan** [comment]
Yeah, I see, the key is that first lib - is a global lib, current implementation checks only module-level dependencies/libs

**Comment by EBatTiVo** [comment]
> Yeah, I see, the key is that first lib - is a global lib, current implementation checks only module-level dependencies/libs

The current implementation checks global libs as well as module libs.  It even removes module level libs (that were automatically added) if they are added to global libs.  At least, that's the way it's /supposed/ to work.
